executor/point_get.go

```diff
@@ -181,15 +181,17 @@ func (e *PointGetExecutor) lockKeyIfNeeded(ctx context.Context, key []byte) erro
 		seVars := e.ctx.GetSessionVars()
 		lockCtx := newLockCtx(seVars, e.lockWaitTime)
 		lockCtx.ReturnValues = true
-		lockCtx.Values = map[string][]byte{}
+		lockCtx.Values = map[string]kv.ReturnedValue{}
 		err := doLockKeys(ctx, e.ctx, lockCtx, key)
 		if err != nil {
 			return err
 		}
 		lockCtx.ValuesLock.Lock()
 		defer lockCtx.ValuesLock.Unlock()
 		for key, val := range lockCtx.Values {
-			seVars.TxnCtx.SetPessimisticLockCache(kv.Key(key), val)
+			if !val.AlreadyLocked {
```

Comment:
Will this lead to reading incorrect data?

begin

point get lock key1,  the lock operation succeeded, but the following operations of this statement failed  for some reasons, the value is cached to txn middle cache, but the statement is async rollbacked

batch point get lock again,  reading values from the middle cache, but the value is invalid

Or shoud we just let pessimistic lock always return value in the pessimistic lock response

Reply:
The pessimistic lock response does always return value, but TiDB would send lock request if it's already locked.

If the key is successfully locked, then we will never do PessimisticRollback on this key even if there are other errors for the statement.